MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2615,572 +2579,2178 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
             return string.Format(CultureInfo.InvariantCulture, filter, widthParam, heightParam);
         }
 
-        /// <summary>
-        /// Gets the output size parameter.
-        /// </summary>
-        /// <param name="state">Encoding state.</param>
-        /// <param name="options">Encoding options.</param>
-        /// <param name="outputVideoCodec">Video codec to use.</param>
-        /// <returns>The output size parameter.</returns>
-        public string GetOutputSizeParam(
-            EncodingJobInfo state,
-            EncodingOptions options,
-            string outputVideoCodec)
+        public static string GetSwDeinterlaceFilter(EncodingJobInfo state, EncodingOptions options)
         {
-            string filters = GetOutputSizeParamInternal(state, options, outputVideoCodec);
-            return string.IsNullOrEmpty(filters) ? string.Empty : " -vf \"" + filters + "\"";
+            var doubleRateDeint = options.DeinterlaceDoubleRate && (state.VideoStream?.AverageFrameRate ?? 60) <= 30;
```

Comment:
Redundant. `<=` comparisons with null return false.
```suggestion
            var doubleRateDeint = options.DeinterlaceDoubleRate && state.VideoStream?.AverageFrameRate <= 30;
```